nexus/user.ts

```diff
@@ -148,11 +148,25 @@ const UserMutations = extendType({
         email: stringArg({ required: true }),
         password: stringArg({ required: true }),
       },
-      resolve: async (_parent, args, ctx: any) => {
+      resolve: async (_parent, args, ctx) => {
         if (!args.handle.match(/^[a-zA-Z0-9_-]+$/)) {
           throw new Error('Invalid handle')
         }
 
+        let existingUser = await ctx.db.user.findUnique({
+          where: {
+            handle: args.handle,
+          },
+        })
+        if (existingUser) throw new Error("Handle is already in use")
+
+        existingUser = await ctx.db.user.findUnique({
+          where: {
+            email: args.email,
+          },
+        })
+        if (existingUser) throw new Error("Email address is already in use. Please try logging in")
```

Comment:
I'd use `ResolverError` or one of its subclasses from `nexus/errors` here. Maybe `UserInputError` which I think is what we use when changing handles.